link/kprobe.go

```diff
@@ -68,6 +68,36 @@ func (pt probeType) RetprobeBit() (uint64, error) {
 	return uretprobeBit()
 }
 
+func (pt probeType) PerfEventAttr(eventType uint64, symbol, path string, offset, config uint64) (*unix.PerfEventAttr, error) {
+	var sp unsafe.Pointer
+
+	switch pt {
+	case kprobeType:
+		// Create a pointer to a NUL-terminated string for the kernel.
+		ptr, err := unsafeStringPtr(symbol)
+		if err != nil {
+			return nil, err
+		}
+		sp = ptr
+	case uprobeType:
+		ptr, err := unsafeStringPtr(path)
+		if err != nil {
+			return nil, err
+		}
+		sp = ptr
+	}
+
+	attr := unix.PerfEventAttr{
+		Size:   uint32(unsafe.Sizeof(unix.PerfEventAttr{})), // Size is used to create the correct PerfEventAttr struct version
```

Comment:
Why not using one of the `PERF_ATTR_SIZE_VER[0-6]` from `golang.org/x/sys/unix`?

Reply:
I've set it to version 5 (same version strace was returning). With version 6 all tests failed. I wonder if this is something that should get maintained (will it always be safe to use version 5 with different versions of x/sys/unix?)

Reply:
The size of the struct `PerfEventAttr` does not respect the value of the size field. But the kernel reads the size field and expect (at least) this amount of bytes.
And iirc `PERF_ATTR_SIZE_VER6` is larger than `unsafe.Sizeof(unix.PerfEventAttr{})`.

Reply:
👍🏻